packages/db/tests/models/test-granule-model.js

```diff
@@ -215,15 +215,962 @@ test('GranulePgModel.upsert() creates a new running granule', async (t) => {
     status: 'running',
   });
 
-  await granulePgModel.upsert(knex, granule, executionCumulusId);
+  await granulePgModel.upsert({
+    knexOrTrx: knex,
+    granule,
+    executionCumulusId,
+    writeConstraints: true,
+  });
+
+  t.like(
+    await granulePgModel.get(knex, granule),
+    granule
+  );
+});
+
+test('GranulePgModel.upsert() creates a new running granule if writeConstraints is set to false', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'running',
+  });
+
+  await granulePgModel.upsert({
+    knexOrTrx: knex,
+    granule,
+    executionCumulusId,
+    writeConstraints: false,
+  });
+
+  t.like(
+    await granulePgModel.get(knex, granule),
+    granule
+  );
+});
+
+test('GranulePgModel.upsert() will only overwrite allowed fields of a granule if update is to status running, and writeConstraints is set to true and write conditions are met', async (t) => {
```

Comment:
```suggestion
test('GranulePgModel.upsert() will only overwrite allowed fields of a granule if update is to set status to running, and writeConstraints is set to true and write conditions are met', async (t) => {
```
Not sure if my suggestion helps but I found this test statement a bit confusing 😅